### DISCUSSION TITLE
Unable to find component: [...]

### DISCUSSION THREAD

**Discussion by osbornemj** [livewire/livewire#6618] [1. Help] [answered]
I have just upgraded from version 2 to version 3.  I have a blade file `index.blade.php` that contains
`@livewire('admin.position-search')`
In `resources/views/livewire/admin` I have a file `position-search.blade.php`.  In `app/Livewire/Admin` I have `PositionSearch.php`, with the namespace `App\Livewire\Admin`.

When I load `index.blade.php`, I get the error `Unable to find component: [admin.position-search]`.

I have cleared the cache and the view cache.

Everything was working fine in version 2.  I am not using any livewire packages.  Sorry if it is something simple that I am missing.

Is there a way to determine where it is looking for the file, or a way to tell it where to look?

**Comment by blondie63** [comment]
Hi @osbornemj how you fixed this issue?

**Reply by osbornemj** [reply]
Unfortunately I haven't --- I am still using Livewire version 2 for that project.  However, I have noticed the file `livewire-components.php` in the `app\bootstrap\cache` folder, which contains an array that points each blade file used by Livewire to a Livewire component.  Currently the components listed are the ones in the `App\Http\Livewire` directory and sudirectories of that directory.  When I have time I am going to edit that file to point to the directories appropriate for v3 of Livewire and try again to upgrade.  I don't know whether that is the problem, but it looks like it might be.  Do you see such a file in your system?

**Reply by blondie63** [reply]
Hi @osbornemj I've found the solution:
Save your component and delete it
Upgrade Livewire to latest 3.4.9
Create a new one with same name:
php artisan make:livewire [your_name] 
and you can see the new namespace is "namespace App\Livewire;" with same path

Then copy your previous code without namespace
Then use this wizard:
php artisan livewire:upgrade -vvv

**Reply by osbornemj** [reply, marked as answer]
The solution suggested by @blondie63 didn't work.  Instead, more than a year after I encountered the problem I have found the issue: the livewire upgrade routine did not set the permission on the app/Livewire directory correctly.  The solution is contained in the last post on 
https://github.com/livewire/livewire/discussions/7668
(many thanks, @NBA707!).

**Reply by joshvand** [reply]
Another reason, maybe due to the livewire config file being slightly incorrect. Specifically with the class_namespace 

`V2:  'class_namespace' => 'App\\Http\\Livewire'`
`V3: 'class_namespace' => 'App\\Livewire'`

The upgrade guide does make mention of changes to the config file and to the namespace change, but not sure if the automated tool does. Not everyone will choose to make that change of course. 

[Here is the default V3 config file ](https://github.com/livewire/livewire/blob/main/config/livewire.php)

**Reply by Ridz010** [reply]
Thanks, you help me for the same problems
